frontend/src/routes/Embed.tsx

```diff
@@ -78,12 +58,85 @@ export const EmbedVideoRoute = makeRoute({
     },
 });
 
+/** Direct link to video with Opencast ID: `/!v/:<ocid>` */
+export const EmbedOpencastVideoRoute = makeRoute({
+    url: (args: { ocID: string }) => `/~embed/!v/:${args.ocID}`,
+    match: url => {
+        const regex = new RegExp("^/~embed/!v/:([^/]+)$", "u");
+        const matches = regex.exec(url.pathname);
+        if (!matches) {
+            return null;
+        }
+
+        const query = graphql`
+            query EmbedDirectOpencastQuery($id: String!) {
+                event: eventByOpencastId(id: $id)  { ... EmbedEventData }
+            }
+            `;
+
+        const videoId = decodeURIComponent(matches[1]);
+        const queryRef = loadQuery<EmbedDirectOpencastQuery>(query, { id: videoId });
+
+        return {
+            render: () => <ErrorBoundary>
+                <Suspense fallback={
+                    <PlayerPlaceholder>
+                        <Spinner css={{
+                            "& > circle": {
+                                stroke: "white",
+                            },
+                        }} />
+                    </PlayerPlaceholder>
+                }>
+                    <PlayerContextProvider>
+                        <Embed query={query} queryRef={queryRef} />
+                    </PlayerContextProvider>
+                </Suspense>
+            </ErrorBoundary>,
```

Comment:
The whole JSX expression is exactly the same for both routes. Can you extract that somehow (e.g. a component) to reduce the duplication here?